applications/GeoMechanicsApplication/custom_constitutive/thermal_dispersion_law.cpp

```diff
@@ -0,0 +1,80 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//                   Gennady Markelov
+//
+
+#include "custom_constitutive/thermal_dispersion_law.hpp"
+#include "custom_retention/retention_law_factory.h"
+#include "geo_mechanics_application_variables.h"
+#include <iostream>
+
+namespace Kratos {
+
+GeoThermalDispersionLaw::GeoThermalDispersionLaw(std::size_t NumberOfDimensions)
+    : mNumberOfDimensions{NumberOfDimensions}
+{
+}
```

Comment:
Shouldn't we need to add a check on `mNumberOfDimensions` here? Probably only the values 2 and 3 make sense here.
```suggestion
{
    KRATOS_ERROR_IF(mNumberOfDimensions < 2 || mNumberOfDimensions > 3) << "Got invalid number of dimensions: " << mNumberOfDimensions << std::endl;
}
```

Reply:
good idea!